pkg/kubelet/rkt/rkt.go

```diff
@@ -425,6 +436,77 @@ func (r *Runtime) makePodManifest(pod *api.Pod, pullSecrets []api.Secret) (*appc
 	var globalPortMappings []kubecontainer.PortMapping
 	manifest := appcschema.BlankPodManifest()
 
+	listReq := &rktapi.ListPodsRequest{
+		Filter: &rktapi.PodFilter{
+			States: []rktapi.PodState{
+				//TODO: In the future some pods can remain running after some apps exit: https://github.com/appc/spec/pull/500
+				rktapi.PodState_POD_STATE_RUNNING,
+				rktapi.PodState_POD_STATE_EXITED,
+				rktapi.PodState_POD_STATE_DELETING,
+				rktapi.PodState_POD_STATE_GARBAGE,
+			},
+			Annotations: []*rktapi.KeyValue{
+				&rktapi.KeyValue{
+					Key:   k8sRktKubeletAnno,
+					Value: k8sRktKubeletAnnoValue,
+				},
+				&rktapi.KeyValue{
+					Key:   k8sRktUIDAnno,
+					Value: string(pod.UID),
+				},
+			},
+		},
+	}
+	listResp, err := r.apisvc.ListPods(context.Background(), listReq)
+	if err != nil {
+		return nil, fmt.Errorf("couldn't list pods: %v", err)
+	}
+
+	restartCount := 0
+	for _, rktpod := range listResp.Pods {
+		manifest := appcschema.PodManifest{}
+		err = json.Unmarshal(rktpod.Manifest, manifest)
+		if err != nil {
+			return nil, err
+		}
+		if countString, ok := manifest.Annotations.Get(k8sRktRestartCountAnno); ok {
+			num, err := strconv.Atoi(countString)
+			if err != nil {
+				continue
+			}
+			if num+1 > restartCount {
+				restartCount = num + 1
+			}
+		}
+	}
+
+	manifest.Annotations = append(manifest.Annotations, []appctypes.Annotation{
```

Comment:
To be pedantic - you should check for any existing annotations by these names, since the spec says they must be [unique](https://github.com/appc/spec/blame/master/spec/pods.md#L181):  
> Annotation names must be unique within the list.